src/client/connection.ts

```diff
@@ -57,35 +55,49 @@ export const INTENTIONAL_DISCONNECT_CODE = 4000
 type WebsocketState = 0 | 1 | 2 | 3
 
 function getAgent(url: string, config: ConnectionOptions): Agent | undefined {
-  // TODO: replace deprecated method
-  if (config.proxy != null) {
-    const parsedURL = parseURL(url)
-    const parsedProxyURL = parseURL(config.proxy)
-    const proxyOverrides = _.omitBy(
-      {
-        secureEndpoint: parsedURL.protocol === 'wss:',
-        secureProxy: parsedProxyURL.protocol === 'https:',
-        auth: config.proxyAuthorization,
-        ca: config.trustedCertificates,
-        key: config.key,
-        passphrase: config.passphrase,
-        cert: config.certificate,
-      },
-      (value) => value == null,
-    )
-    const proxyOptions = { ...parsedProxyURL, ...proxyOverrides }
-    let HttpsProxyAgent
-    try {
-      // eslint-disable-next-line max-len -- Long eslint-disable-next-line TODO: figure out how to make this nicer
-      // eslint-disable-next-line import/max-dependencies, @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-require-imports, node/global-require, global-require, -- Necessary for the `require`
-      HttpsProxyAgent = require('https-proxy-agent')
-    } catch (_error) {
-      throw new Error('"proxy" option is not supported in the browser')
-    }
-    // eslint-disable-next-line @typescript-eslint/consistent-type-assertions, @typescript-eslint/no-unsafe-call -- Necessary
-    return new HttpsProxyAgent(proxyOptions) as unknown as Agent
+  if (config.proxy == null) {
+    return undefined
+  }
+
+  const parsedURL = new URL(url)
+  const parsedProxyURL = new URL(config.proxy)
+
+  const proxyOptions = _.omitBy(
```

Comment:
Why does this have more options than the old version?

Reply:
Moved the options from `parsedProxyURL` into the same object as the proxyOptions object.